modules/client/src/allowlistVoting/internal/methods.ts

```diff
@@ -0,0 +1,470 @@
+
+  constructor(allowlistVoting: AllowlistVoting) {
+    this.allowlistVoting = allowlistVoting;
+  }
+
+  /**
+   * Returns the hash for the MODIFY_ALLOWLIST_PERMISSION permission
+   *
+   * @return {*}  {Promise<string>}
+   * @memberof AllowlistVotingMethods
+   */
+  public MODIFY_ALLOWLIST_PERMISSION_ID(): Promise<string> {
+    return this.allowlistVoting
+      .getPluginInstanceWithSigner()
+      .MODIFY_ALLOWLIST_PERMISSION_ID();
+  }
+
+  /**
+   * Returns the PCT_BASE
+   *
+   * @return {*}  {Promise<number>}
+   * @memberof AllowlistVotingMethods
+   */
+  public async PCT_BASE(): Promise<number> {
+    return (
+      await this.allowlistVoting.getPluginInstanceWithSigner().PCT_BASE()
+    ).toNumber();
+  }
+
+  /**
+   * Returns the hash for the SET_CONFIGURATION_PERMISSION permission
+   *
+   * @return {*}  {Promise<string>}
+   * @memberof AllowlistVotingMethods
+   */
+  public SET_CONFIGURATION_PERMISSION_ID(): Promise<string> {
+    return this.allowlistVoting
+      .getPluginInstanceWithSigner()
+      .SET_CONFIGURATION_PERMISSION_ID();
+  }
+
+  /**
+   * Returns the hash for the UPGRADE_PLUGIN_PERMISSION permission
+   *
+   * @return {*}  {Promise<string>}
+   * @memberof AllowlistVotingMethods
+   */
+  public UPGRADE_PLUGIN_PERMISSION_ID(): Promise<string> {
+    return this.allowlistVoting
+      .getPluginInstanceWithSigner()
+      .UPGRADE_PLUGIN_PERMISSION_ID();
+  }
+
+  /**
+   * Adds new allowed users to the plugin.
+   * Returns a generator with 2 steps
+   *
+   * @param {string[]} _users
+   * @return {*}  {AsyncGenerator<VoteStepsValue>}
+   * @memberof AllowlistVotingMethods
+   */
+  public async *addAllowedUsers(
+    _users: string[]
+  ): AsyncGenerator<VoteStepsValue> {
+    const tx = await this.allowlistVoting
+      .getPluginInstanceWithSigner()
+      .addAllowedUsers(_users);
+    yield {
+      key: Steps.PENDING,
+      txHash: tx.hash,
+    };
+    await tx.wait();
+    yield {
+      key: Steps.DONE,
+    };
+  }
+
+  /**
+   * Returns the amount of allowed users at a given block
+   *
+   * @param {BigInt} blockNumber
+   * @return {*}  {Promise<BigInt | number>}
+   * @memberof AllowlistVotingMethods
+   */
+  public async allowedUserCount(blockNumber: BigInt | number): Promise<BigInt> {
+    return (
+      await this.allowlistVoting
+        .getPluginInstanceWithSigner()
+        .allowedUserCount(BigNumber.from(blockNumber))
+    ).toBigInt();
+  }
+
+  /**
+   * Checks if a proposal can be executed
+   *
+   * @param {BigInt | number} _proposalId
+   * @return {*}  {Promise<boolean>}
+   * @memberof AllowlistVotingMethods
+   */
+  public canExecute(_proposalId: BigInt | number): Promise<boolean> {
+    return this.allowlistVoting
+      .getPluginInstanceWithSigner()
+      .canExecute(BigNumber.from(_proposalId));
+  }
+
+  /**
+   * Checks if a voter can vote on a proposal
+   *
+   * @param {BigInt | number} _proposalId
+   * @param {string} _voter
```

Comment:
Same as before for the `proposalId`, also we plan on start using `yup` to validate the inputs coming from the user so I believe this needs to become a params type or Interface, otherwise when we introduce `yup` this will mean grouping this into a yup schema and posibly breaking the ui